From 5e87c5a6f53b1bbb46d4cadad706179ca6714fc1 Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Thu, 28 Feb 2019 08:46:04 +0800
Subject: [PATCH] media: rockchip: isp1: stop mipi with shutdown lane

isp_mipi_ctrl[8:11] control synopsys dphy's SHUTDOWNZ
and RSTZ. stop mipi and dphy together.

Change-Id: Ief50ab81f5f960e414a880e43b772ee47752742b
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rockchip/isp1/rkisp1.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/rockchip/isp1/rkisp1.c b/drivers/media/platform/rockchip/isp1/rkisp1.c
index 6c280728e265..cb8de6b96446 100644
--- a/drivers/media/platform/rockchip/isp1/rkisp1.c
+++ b/drivers/media/platform/rockchip/isp1/rkisp1.c
@@ -596,7 +596,9 @@ static int rkisp1_isp_stop(struct rkisp1_device *dev)
 		writel(0, base + CIF_ISP_CSI0_CTRL0);
 	} else {
 		val = readl(base + CIF_MIPI_CTRL);
+		val = val & (~CIF_MIPI_CTRL_SHUTDOWNLANES(0xf));
 		writel(val & (~CIF_MIPI_CTRL_OUTPUT_ENA), base + CIF_MIPI_CTRL);
+		udelay(20);
 	}
 	/* stop ISP */
 	val = readl(base + CIF_ISP_CTRL);
-- 
2.35.3

